src/Components/Server/src/DependencyInjection/RazorComponentsServiceCollectionExtensions.cs

```diff
@@ -1,115 +1,57 @@
+            services.AddSignalR().AddMessagePackProtocol();
 
-        private static IServiceCollection AddRazorComponentsCore(
-            IServiceCollection services,
-            Type startupType)
-        {
             AddStandardRazorComponentsServices(services);
 
-            if (startupType != null)
-            {
-                // Call TStartup's ConfigureServices method immediately
-                var startup = Activator.CreateInstance(startupType);
-                var wrapper = new ConventionBasedStartup(startup);
-                wrapper.ConfigureServices(services);
-
-                // Configure the circuit factory to call a startup action when each
-                // incoming connection is established. The startup action is "call
-                // TStartup's Configure method".
-                services.Configure<DefaultCircuitFactoryOptions>(circuitFactoryOptions =>
-                {
-                    var endpoint = ComponentsHub.DefaultPath; // TODO: allow configuring this
-                    if (circuitFactoryOptions.StartupActions.ContainsKey(endpoint))
-                    {
-                        throw new InvalidOperationException(
-                            "Multiple Components app entries are configured to use " +
-                            $"the same endpoint '{endpoint}'.");
-                    }
-
-                    circuitFactoryOptions.StartupActions.Add(endpoint, builder =>
-                    {
-                        wrapper.Configure(builder, builder.Services);
-                    });
-                });
-            }
-
-            return services;
-        }
-
-        private static void AddStandardRazorComponentsServices(IServiceCollection services)
-        {
             // Here we add a bunch of services that don't vary in any way based on the
             // user's configuration. So even if the user has multiple independent server-side
             // Components entrypoints, this lot is the same and repeated registrations are a no-op.
+            services.TryAddEnumerable(ServiceDescriptor.Singleton<IPostConfigureOptions<StaticFileOptions>, ConfigureStaticFilesOptions>());
             services.TryAddSingleton<CircuitFactory, DefaultCircuitFactory>();
-            services.TryAddScoped<ICircuitAccessor, DefaultCircuitAccessor>();
             services.TryAddScoped(s => s.GetRequiredService<ICircuitAccessor>().Circuit);
+#pragma warning disable CS0618 // Type or member is obsolete
             services.TryAddScoped<IJSRuntimeAccessor, DefaultJSRuntimeAccessor>();
-            services.TryAddScoped(s => s.GetRequiredService<IJSRuntimeAccessor>().JSRuntime);
-            services.TryAddScoped<IUriHelper, RemoteUriHelper>();
+#pragma warning restore CS0618 // Type or member is obsolete
+            services.TryAddScoped<ICircuitAccessor, DefaultCircuitAccessor>();
+            services.AddScoped<IComponentPrerrenderer, CircuitPrerrenderer>();
```

Comment:
It's worth a comment here to point out that this is intentional.  `Add` vs `TryAdd`. 

Do you think that we should take any level of care to try and only replace if it's the default implementation? I don't have much experience with the impact of replacing services because we don't do it many places.

Reply:
Yes.

Identity does it all over the place. I haven't seen it cause a lot of problems, and I wouldn't expect people to come up with different implementations for components. We essentially require our abstractions when we are using circuits.

Reply:
Added a comment here to point it out.